netebpfext/net_ebpf_ext_hook_provider.c

```diff
@@ -171,11 +171,11 @@ _net_ebpf_extension_detach_client_completion(_In_ DEVICE_OBJECT* device_object,
     // Wait for any in progress callbacks to complete.
     _ebpf_ext_attach_wait_for_rundown(&hook_client->rundown);
 
+    IoFreeWorkItem(work_item);
```

Comment:
This change is needed because in user mode tests, due to race condition, memory leak detector marks this memory as leaked before the memory is freed.